jabra/callstats-jabra-shim.js

```diff
@@ -0,0 +1,195 @@
+/*! callstats Jabra SHIM version = 1.0.0 */
+
+(function (global) {
+  var CallstatsJabraShim = function() {
+    let isTxSpeechStarted = false;
+    let isRxSpeechStarted = false;
+    let isCrossTalkStarted = false;
+
+    function sendJabraActiveState(txSpeech, rxSpeech) {
+      if (txSpeech === undefined && rxSpeech === undefined) {
+        return;
+      }
+
+      if (!isTxSpeechStarted && txSpeech) {
+        isTxSpeechStarted = true;
+      } else if (txSpeech === false && isTxSpeechStarted) {
+        isTxSpeechStarted = false;
+        if (!isCrossTalkStarted) {
+          sendCustomEvent('agentSpeakingStop');
+        }
+      }
+
+      if (!isRxSpeechStarted && rxSpeech) {
+        isRxSpeechStarted = true;
+      } else if (rxSpeech === false && isRxSpeechStarted) {
+        isRxSpeechStarted = false;
+        if (!isCrossTalkStarted) {
+          sendCustomEvent('contactSpeakingStop');
+        }
+      }
+
+      if (isTxSpeechStarted && isRxSpeechStarted) {
+        isCrossTalkStarted = true;
+      } else if (isCrossTalkStarted && (isTxSpeechStarted || isRxSpeechStarted)) {
```

Comment:
Just wondering - what will happen in use case like
`both talking -> cross talk`
Now if both stop talking at the same time then cross talk will be still `true`, right. Am I missing something ?